streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -157,10 +157,9 @@ int addRawRecords(final TopicPartition partition, final Iterable<ConsumerRecord<
     }
 
     /**
-     * Return the timestamp of this partition group as the smallest
-     * partition timestamp among all its partitions
+     * Return the stream-time of this partition group defined as the largest timestamp seen across all partitions
      */
-    public long timestamp() {
+    public long streamTime() {
```

Comment:
Should we rename this this `taskTime()`? (Just a thought).

Reply:
I'm in favor of that in theory -- but, then do we also rename `maybePuncuateStreamTime` ? Do we also deprecate `PunctuationType.STREAM_TIME` in favor of `PunctuationType.TASK_TIME`? Task time does seem more appropriate but I'm hesitant to mix terminology ...

Reply:
Let's just keep it as `streamTime` for now.